CONTRIBUTING.md

```diff
@@ -15,42 +14,103 @@
+- `commands` is the package where the WildFly CLI abstractions are being put. This package should also include just the operations, that are present in the `jboss-cli.sh` but are not available via REST API (commands such as `data-source add ...`, `module add ...`)
 
-* `resources` package serves mostly to build Java deployments/JARs (such as JDBC drivers), basically anything related to the WildFly Java world.
+- `resources` package serves mostly to build Java deployments/JARs (such as JDBC drivers), basically anything related to the WildFly Java world.
 
 ## How to write tests
 
 ### Differentiate between NodeJS runtimes
 
 When developing in Cypress, you need to understand two different JavaScript runtimes
 
-* The one Cypress bundles into the browser under test
-* System wide JavaScript
+- The one Cypress bundles into the browser under test
+- System wide JavaScript
+
+Since using TestContainers to launch containers shouldn't be bundled into browser under test, starting the WildFly container is being done using Cypress arbitrary Node code running handle - [`cy.task`](https://docs.cypress.io/api/commands/task), use
 
-Since using TestContainers to launch containers shouldn't be bundled into browser under test, starting the WildFly container is being done using Cypress arbitrary Node code running handle - [`cy.task`](https://docs.cypress.io/api/commands/task), use 
 ```
 cy.startWildFlyContainer
 ```
+
 method, which is a wrapper for
+
 ```
 cy.task("start:wildfly:container")
 ```
-to start the container. This method/task also automatically sets the deployed HAL container endpoint in the `allowed-origins` attribute. Both of the approaches would return the management endpoint of the deployed WildFly instance.
+
+to start the container. Note that despite starting the web server being marked as an anti-pattern, starting the WildFly server clean has its benefits in terms of a clean environment per test and also it introduces means for test parallelization (we know what we're doing :slightly_smiling_face:). This method/task also automatically sets the deployed HAL container endpoint in the `allowed-origins` attribute. Both of the approaches would return the management endpoint of the deployed WildFly instance.
 
 ### Communicating with WildFly management interface
 
 Communication with the WildFly management interface is also being done using Cypress task, `execute:cli` to be precise. A better example of running `/system-property=to-add:add(value=myVal)` in JBoss CLI in Berg using `cy.task`
-would be 
+would be
+
 ```
 cy.task("execute:cli", {
       managementApi: `${managementEndpoint}/management`,
       operation: "add",
       address: ["system-property", "to-add"],
       value: "myVal"
     });
-```
\ No newline at end of file
+```
+
+### Test template
+
+To start developing tests, you can copy paste following example into your `packages/testsuite/cypress/e2e/test-to-be-added.cy.ts`
```

Comment:
Should any new test have it's own new file? Are there some naming conventions? How to make sure to run my new tests in regular testing?

Reply:
> How to make sure to run my new tests in regular testing?

I appreciate you have added basic commands for test execution, but my question is rather about stuff like https://github.com/hal/berg/blob/main/.github/workflows/scheduled-run-all-tests-workflow.yaml#L12

Reply:
@OndrejKotek I made a bullet list of the intentions + added section about GH workflows, can this thread be resolved?